daemon/oci_windows.go

```diff
@@ -25,11 +25,31 @@ func (daemon *Daemon) createSpec(c *container.Container) (*specs.Spec, error) {
 	// In base spec
 	s.Hostname = c.FullHostname()
 
+	if err := daemon.setupSecretDir(c); err != nil {
+		return nil, err
+	}
+
 	// In s.Mounts
 	mounts, err := daemon.setupMounts(c)
 	if err != nil {
 		return nil, err
 	}
+
+	if !c.HasBeenStartedBefore && len(c.SecretReferences) > 0 {
+		if err := daemon.Mount(c); err != nil {
```

Comment:
Again, as per offline, if we can avoid this and not rely on reference counting, that would be far better and more explicit. Note that as also mentioned, this should be co-ordinated with the configs PR